flips/20220811-fungible-tokens-metadata.md

```diff
@@ -0,0 +1,210 @@
+            self.externalURL = externalURL
+            self.socials = socials
+        }
+    }
+```
+
+```cadence
+    /// View to expose the information needed store and interact with a FT vault.
+    /// This can be used by applications to setup a FT vault with proper 
+    /// storage and public capabilities.
+    ///
+    pub struct FTVaultData {
+        /// Path in storage where this FT vault is recommended to be stored.
+        pub let storagePath: StoragePath
+
+        /// Public path which must be linked to expose public capabilities of
+        /// this FT, including standard FT interfaces and metadataviews 
+        /// interfaces
+        pub let publicPath: PublicPath
+
+        /// Private path which should be linked to expose the provider
+        /// capability to withdraw funds from the vault
+        pub let providerPath: PrivatePath
+
+        /// Type that should be linked at the aforementioned public path. This 
+        /// is normally a restricted type with many interfaces. Notably the 
+        /// `FT.Balance`, `FT.Receiver`, and `FungibleMetadataViews.Resolver` 
+        /// interfaces are required.
+        pub let publicLinkedType: Type
+
+        /// Type that should be linked at the aforementioned private path. This 
+        /// is normally a restricted type with at a minimum the `FT.Provider` 
+        /// interface
+        pub let providerLinkedType: Type
+
+        /// Function that allows creation of an empty FT vault that is intended
+        /// to store the funds.
+        pub let createEmptyVault: ((): @FungibleToken.Vault)
+
+        init(
+            storagePath: StoragePath,
+            publicPath: PublicPath,
+            providerPath: PrivatePath,
+            publicCollection: Type,
+            publicLinkedType: Type,
+            providerLinkedType: Type,
+            createEmptyVaultFunction: ((): @FungibleToken.Vault)
+        ) {
+            pre {
+                publicLinkedType.isSubtype(of: Type<&{FungibleToken.Receiver, FungibleToken.Balance, Resolver}>()): "Public type must include FungibleToken.Receiver, FungibleToken.Balance and FungibleMetadataViews.Resolver interfaces."
+                providerLinkedType.isSubtype(of: Type<&{FungibleToken.Provider, Resolver}>()): "Provider type must include NonFungibleToken.Provider and FungibleMetadataViews.Resolver interface."
+            }
+            self.storagePath=storagePath
+            self.publicPath=publicPath
+            self.providerPath = providerPath
+            self.publicLinkedType=publicLinkedType
+            self.providerLinkedType = providerLinkedType
+            self.createEmptyVault = createEmptyVaultFunction
+        }
+    }
+```
+
+In order to be able to use the `FTDisplay` we will need to re-use or re-implement (depending on if this ends up being part of the original `MetadataViews.cdc` contract or having its own contract) the following views: `ExternalURL`, `File`, `HTTPFile`, `IPFSFile`, `Media` and `Medias`. 
```

Comment:
I think we can import those views from the original contract. no need to re-implement them

Reply:
Yeah totally agree